File: test_connection.py
import requests
import json

# URL de l'API
url = "http://localhost:5000/register"

# Données de test
data = {
    "firstName": "Test",
    "lastName": "User",
    "email": "test@example.com",
    "parentalEmail": "parent@example.com",
    "phoneNumber": "0123456789",
    "parentalPhoneNumber": "0987654321",
    "password": "password123",
    "confirmPassword": "password123"
}

headers = {
    'Content-Type': 'application/json'
}

try:
    # Envoyer la requête
    response = requests.post(url, headers=headers, data=json.dumps(data))
    
    # Afficher la réponse
    print("Status Code:", response.status_code)
    print("Response:", response.text)
    
except requests.exceptions.RequestException as e:
    print("Erreur de connexion:", str(e))

File: LeoniApp/TestConnection.js
// Test simple pour vérifier la connexion
const testConnection = async () => {
  try {
    console.log('Testing connection to backend...');
    
    const response = await fetch('http://localhost:5000/login', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        email: 'y@gmail.com',
        password: 'yyyyyy'
      })
    });
    
    console.log('Response status:', response.status);
    console.log('Response headers:', response.headers);
    
    const data = await response.json();
    console.log('Response data:', data);
    
    if (data.success) {
      console.log('✅ Connexion réussie!');
      console.log('Token:', data.token);
      console.log('User:', data.user);
    } else {
      console.log('❌ Connexion échouée:', data.message);
    }
  } catch (error) {
    console.error('❌ Erreur de connexion:', error);
  }
};

// Exécuter le test
testConnection();
